File: Data Structures & Algorithms/depth-of-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public int maxDepth(TreeNode root) {
        Queue<TreeNode> q = new LinkedList<>();
        if (root != null) {
            q.add(root);
        }

        int level = 0;
        // while q isnt empty
        while (!q.isEmpty()) {
            int size = q.size();
            // traverse size of tree
            for (int i = 0; i < size; i++) {
                // get node sitting in queue
                TreeNode node = q.poll();
                // if node sitting in queue is not null add
                if (node.left != null) {
                    q.add(node.left);
                }
                // check independently
                if (node.right != null) {
                    q.add(node.right);
                }
            }
            // add to level
            level++;
        }
        return level;
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.py
from collections import deque
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if root is None:
            return []
        res = []
        queue = deque([root])
        while len(queue) > 0:
            level = []
            for i in range(len(queue)):
                curr = queue.popleft() # pop the leftmost node
                level.append(curr.val)
                if curr.left is not None:
                    queue.append(curr.left)
                if curr.right is not None:
                    queue.append(curr.right)
            res.append(level)
        return res